From: andi34 <skate4life@gmx.de>
Date: Sat, 12 Jul 2014 11:55:20 +0200
Subject: [PATCH] arch/arm/mach-omap2/Kconfig.mid: revert JB Update 1/3

diff --git a/arch/arm/mach-omap2/Kconfig.mid b/arch/arm/mach-omap2/Kconfig.mid
old mode 100755
new mode 100644
index 66c4d4c..3d3b7f1
--- a/arch/arm/mach-omap2/Kconfig.mid
+++ b/arch/arm/mach-omap2/Kconfig.mid
@@ -2,16 +2,3 @@ config MACH_SAMSUNG_GOKEY
 	bool "Samsung Gokey (YP-GH1) Board"
 	default n
 	select MACH_OMAP4_SAMSUNG
-	select FW_SAMSUNG_GOKEY_TSP
-
-config FW_SAMSUNG_GOKEY_TSP
-	bool
-	default n
-
-config MP3_LP_MODE
-	bool "mp3 LP"
-	default n
-	help
-	  to reduce mp3 current consumption, turn down cpu1 and limit
-	  cpu frequency when the device is in earlysuspend mode MP3
-	  playback.
\ No newline at end of file
