From: andi34 <skate4life@gmx.de>
Date: Sat, 12 Jul 2014 11:55:20 +0200
Subject: [PATCH] arch/arm/mach-omap2/board-espresso10-muxset-r00.c: revert JB Update 1/3

diff --git a/arch/arm/mach-omap2/board-espresso10-muxset-r00.c b/arch/arm/mach-omap2/board-espresso10-muxset-r00.c
index a2b1cd0..c571baa 100644
--- a/arch/arm/mach-omap2/board-espresso10-muxset-r00.c
+++ b/arch/arm/mach-omap2/board-espresso10-muxset-r00.c
@@ -27,49 +27,49 @@ static struct omap_muxtbl muxtbl[] __initdata = {
 	OMAP4_MUXTBL(OMAP4_MUXTBL_DOMAIN_CORE,
 		     GPMC_AD0,
 		     OMAP_MUX_MODE1 | OMAP_PIN_INPUT_PULLUP |
-		     OMAP_PIN_OFF_NONE,
+		     OMAP_PIN_OFF_INPUT_PULLDOWN,
 		     OMAP_MUXTBL_NO_GPIO, "NAND_D(0)"),
 	/* [-----] gpmc_ad1.sdmmc2_dat1 - NAND_D(1) */
 	OMAP4_MUXTBL(OMAP4_MUXTBL_DOMAIN_CORE,
 		     GPMC_AD1,
 		     OMAP_MUX_MODE1 | OMAP_PIN_INPUT_PULLUP |
-		     OMAP_PIN_OFF_NONE,
+		     OMAP_PIN_OFF_INPUT_PULLDOWN,
 		     OMAP_MUXTBL_NO_GPIO, "NAND_D(1)"),
 	/* [-----] gpmc_ad2.sdmmc2_dat2 - NAND_D(2) */
 	OMAP4_MUXTBL(OMAP4_MUXTBL_DOMAIN_CORE,
 		     GPMC_AD2,
 		     OMAP_MUX_MODE1 | OMAP_PIN_INPUT_PULLUP |
-		     OMAP_PIN_OFF_NONE,
+		     OMAP_PIN_OFF_INPUT_PULLDOWN,
 		     OMAP_MUXTBL_NO_GPIO, "NAND_D(2)"),
 	/* [-----] gpmc_ad3.sdmmc2_dat3 - NAND_D(3) */
 	OMAP4_MUXTBL(OMAP4_MUXTBL_DOMAIN_CORE,
 		     GPMC_AD3,
 		     OMAP_MUX_MODE1 | OMAP_PIN_INPUT_PULLUP |
-		     OMAP_PIN_OFF_NONE,
+		     OMAP_PIN_OFF_INPUT_PULLDOWN,
 		     OMAP_MUXTBL_NO_GPIO, "NAND_D(3)"),
 	/* [-----] gpmc_ad4.sdmmc2_dat4 - NAND_D(4) */
 	OMAP4_MUXTBL(OMAP4_MUXTBL_DOMAIN_CORE,
 		     GPMC_AD4,
 		     OMAP_MUX_MODE1 | OMAP_PIN_INPUT_PULLUP |
-		     OMAP_PIN_OFF_NONE,
+		     OMAP_PIN_OFF_INPUT_PULLDOWN,
 		     OMAP_MUXTBL_NO_GPIO, "NAND_D(4)"),
 	/* [-----] gpmc_ad5.sdmmc2_dat5 - NAND_D(5) */
 	OMAP4_MUXTBL(OMAP4_MUXTBL_DOMAIN_CORE,
 		     GPMC_AD5,
 		     OMAP_MUX_MODE1 | OMAP_PIN_INPUT_PULLUP |
-		     OMAP_PIN_OFF_NONE,
+		     OMAP_PIN_OFF_INPUT_PULLDOWN,
 		     OMAP_MUXTBL_NO_GPIO, "NAND_D(5)"),
 	/* [-----] gpmc_ad6.sdmmc2_dat6 - NAND_D(6) */
 	OMAP4_MUXTBL(OMAP4_MUXTBL_DOMAIN_CORE,
 		     GPMC_AD6,
 		     OMAP_MUX_MODE1 | OMAP_PIN_INPUT_PULLUP |
-		     OMAP_PIN_OFF_NONE,
+		     OMAP_PIN_OFF_INPUT_PULLDOWN,
 		     OMAP_MUXTBL_NO_GPIO, "NAND_D(6)"),
 	/* [-----] gpmc_ad7.sdmmc2_dat7 - NAND_D(7) */
 	OMAP4_MUXTBL(OMAP4_MUXTBL_DOMAIN_CORE,
 		     GPMC_AD7,
 		     OMAP_MUX_MODE1 | OMAP_PIN_INPUT_PULLUP |
-		     OMAP_PIN_OFF_NONE,
+		     OMAP_PIN_OFF_INPUT_PULLDOWN,
 		     OMAP_MUXTBL_NO_GPIO, "NAND_D(7)"),
 	/* [IN---] gpmc_ad8.gpio_32 - TA_nCONNECTED */
 	OMAP4_MUXTBL(OMAP4_MUXTBL_DOMAIN_CORE,
@@ -192,7 +192,7 @@ static struct omap_muxtbl muxtbl[] __initdata = {
 	OMAP4_MUXTBL(OMAP4_MUXTBL_DOMAIN_CORE,
 		     GPMC_NWE,
 		     OMAP_MUX_MODE1 | OMAP_PIN_INPUT_PULLUP |
-		     OMAP_PIN_OFF_NONE,
+		     OMAP_PIN_OFF_INPUT_PULLDOWN,
 		     OMAP_MUXTBL_NO_GPIO, "NAND_CMD"),
 	/* [--OUT] gpmc_nbe0_cle.gpio_59 - IRDA_EN */
 	OMAP4_MUXTBL(OMAP4_MUXTBL_DOMAIN_CORE,
