From 2c2ead353fc86ad7dd2f7f2856a98a9519081ba8 Mon Sep 17 00:00:00 2001
From: andi34 <skate4life@gmx.de>
Date: Sat, 12 Jul 2014 14:56:19 +0200
Subject: [PATCH] arch/arm/mach-omap2/mailbox.c & omap-iommu.c : revert JB Update 1/3

diff --git a/arch/arm/mach-omap2/mailbox.c b/arch/arm/mach-omap2/mailbox.c
index e7e2d37..13bdf9d 100644
--- a/arch/arm/mach-omap2/mailbox.c
+++ b/arch/arm/mach-omap2/mailbox.c
@@ -41,11 +41,7 @@
 #define MBOX_NR_REGS                   2
 #define OMAP4_MBOX_NR_REGS             3
 
-#ifdef CONFIG_ENABLE_GPIO_TO_ALLOW_C2_IN_CAMERA
-#define SET_MPU_CORE_CONSTRAINT		400
-#else
 #define SET_MPU_CORE_CONSTRAINT		10
-#endif
 
 static void __iomem *mbox_base;
 
diff --git a/arch/arm/mach-omap2/omap-iommu.c b/arch/arm/mach-omap2/omap-iommu.c
index 1736020..e49c9d5 100644
--- a/arch/arm/mach-omap2/omap-iommu.c
+++ b/arch/arm/mach-omap2/omap-iommu.c
@@ -54,11 +54,7 @@ static struct iommu_platform_data omap3_devices_data[] = {
 #ifdef CONFIG_ARCH_OMAP4
 
 #define SET_DSP_CONSTRAINT	10
-#ifdef CONFIG_ENABLE_GPIO_TO_ALLOW_C2_IN_CAMERA
-#define SET_MPU_CORE_CONSTRAINT	400
-#else
 #define SET_MPU_CORE_CONSTRAINT	10
-#endif
 
 static struct iommu_platform_data omap4_devices_data[] = {
 	{
