From aa754d5580e211740bf063842707bd32db866306 Mon Sep 17 00:00:00 2001
From: andi34 <skate4life@gmx.de>
Date: Sat, 12 Jul 2014 16:25:18 +0200
Subject: [PATCH] arch/arm/mach-omap2/pm44xx.c: revert JB Update 1/3

---
diff --git a/arch/arm/mach-omap2/pm44xx.c b/arch/arm/mach-omap2/pm44xx.c
index 7cb2a86..02cec01 100644
--- a/arch/arm/mach-omap2/pm44xx.c
+++ b/arch/arm/mach-omap2/pm44xx.c
@@ -1423,8 +1423,7 @@ static void __init omap4_pm_setup_errata(void)
 static int __init omap4_pm_init(void)
 {
 	int ret = 0;
-	struct clockdomain *l3_1_clkdm, *l4wkup, *sdma_clkdm, *l3_init_clkdm, \
-								*l3_gfx_clkdm;
+	struct clockdomain *l3_1_clkdm, *l4wkup;
 	struct clockdomain *ducati_clkdm, *l3_2_clkdm;
 	struct clockdomain *l4_per, *l4_cfg, *ivahd_clkdm;
 
@@ -1492,14 +1491,8 @@ static int __init omap4_pm_init(void)
 	l4wkup = clkdm_lookup("l4_wkup_clkdm");
 	ivahd_clkdm = clkdm_lookup("ivahd_clkdm");
 
-	sdma_clkdm = clkdm_lookup("l3_dma_clkdm");
-	l3_init_clkdm = clkdm_lookup("l3_init_clkdm");
-
-	l3_gfx_clkdm = clkdm_lookup("l3_gfx_clkdm");
-
 	if ((!mpuss_clkdm) || (!emif_clkdm) || (!l3_1_clkdm) || (!l4wkup) ||
-		(!l3_2_clkdm) || (!ducati_clkdm) || (!l4_per) || \
-						(!l4_cfg) || (!l3_gfx_clkdm))
+		(!l3_2_clkdm) || (!ducati_clkdm) || (!l4_per) || (!l4_cfg))
 		goto err2;
 
 	/* if we cannot ever enable static dependency. */
@@ -1518,7 +1511,6 @@ static int __init omap4_pm_init(void)
 		ret |= clkdm_add_wkdep(ducati_clkdm, emif_clkdm);
 		ret |= clkdm_add_wkdep(ivahd_clkdm, emif_clkdm);
 		ret |= clkdm_add_wkdep(mpuss_clkdm, l4wkup);
-		ret |= clkdm_add_wkdep(l3_gfx_clkdm, emif_clkdm);
 
 		if (ret) {
 			pr_err("Failed to add MPUSS -> L3/EMIF, DUCATI -> L3"
@@ -1551,16 +1543,6 @@ static int __init omap4_pm_init(void)
 		ret |= clkdm_add_wkdep(ducati_clkdm, l4_per);
 		ret |= clkdm_add_wkdep(ducati_clkdm, l4_cfg);
 
-		/* Adding all SDMA static dependencies and keeping
-		   only L3_2, before we clear.
-		*/
-		ret |=  clkdm_add_wkdep(sdma_clkdm, emif_clkdm);
-		ret |=  clkdm_add_wkdep(sdma_clkdm, l3_1_clkdm);
-		ret |=  clkdm_add_wkdep(sdma_clkdm, l3_init_clkdm);
-		ret |=  clkdm_add_wkdep(sdma_clkdm, l4wkup);
-		ret |=  clkdm_add_wkdep(sdma_clkdm, l4_per);
-		ret |=  clkdm_add_wkdep(sdma_clkdm, l4_cfg);
-
 		if (ret) {
 			pr_err("Failed to add MPUSS and DUCATI -> "
 			       "L4* and L3_1 wakeup dependency\n");
@@ -1568,18 +1550,6 @@ static int __init omap4_pm_init(void)
 		}
 		pr_info("OMAP4 PM: Static dependency added between"
 			" MPUSS and DUCATI <-> L4_PER/CFG and L3_1.\n");
-		/* clearing all SDMA static dependencies and keeping
-		   only L3_2.
-		*/
-		ret  =  clkdm_del_wkdep(sdma_clkdm, emif_clkdm);
-		ret |=  clkdm_del_wkdep(sdma_clkdm, l3_1_clkdm);
-		ret |=  clkdm_del_wkdep(sdma_clkdm, l3_init_clkdm);
-		ret |=  clkdm_del_wkdep(sdma_clkdm, l4wkup);
-		ret |=  clkdm_del_wkdep(sdma_clkdm, l4_per);
-		ret |=  clkdm_del_wkdep(sdma_clkdm, l4_cfg);
-
-		if (ret)
-			pr_err("Failed to Remove wkdep For SDMA\n");
 	}
 
 	(void) clkdm_for_each(clkdms_setup, NULL);
--
1.9.3


