From aa754d5580e211740bf063842707bd32db866306 Mon Sep 17 00:00:00 2001
From: andi34 <skate4life@gmx.de>
Date: Sat, 12 Jul 2014 16:25:18 +0200
Subject: [PATCH] arch/arm/mach-omap2/sec_logger.c: revert JB Update 1/3

---
diff --git a/arch/arm/mach-omap2/sec_logger.c b/arch/arm/mach-omap2/sec_logger.c
index 98db33a..0e2e29c 100644
--- a/arch/arm/mach-omap2/sec_logger.c
+++ b/arch/arm/mach-omap2/sec_logger.c
@@ -208,14 +208,12 @@ void sec_logger_update_buffer(const char *log_str, int count)
 		sec_klog_buf[len] = '\0';
 	}
 }
-EXPORT_SYMBOL(sec_logger_update_buffer);
 
 void sec_logger_print_buffer(void)
 {
 	if (sec_klog_buf[0])
 		pr_info("%s\n", sec_klog_buf);
 }
-EXPORT_SYMBOL(sec_logger_print_buffer);
 
 static void __init sec_logger_message_init(void)
 {
--
1.9.3


